stonesoup/hypothesiser/distance.py

```diff
@@ -80,11 +79,6 @@ def hypothesise(self, track, detections, timestamp):
 
         # True detection hypotheses
         for detection in detections:
-
-            # Re-evaluate prediction
-            prediction = self.predictor.predict(
-                track.state, timestamp=detection.timestamp)
```

Comment:
This is required in order to handle detentions at different timestamps. If this is an issue for GMPHD, you could check that all time stamps are the same in the `GaussianMixtureHypothesiser` and raise an exception if not.

Reply:
See previous comment about timestamps

Reply:
Added back in. Also added a check in the `GaussianMixtureHypothesier` to make sure all of the detections are at the same timestamp